tests/plugins/mocha.test.support/hooks/hook.js

```diff
@@ -32,24 +33,66 @@ exports.init = (logger, config, cli) => {
 
 	cli.on('build.post.compile', async (builder, done) => {
 		if (builder.platformName === 'android') {
-			await wakeDevices(logger, builder).catch(e => logger.warn(`could not wake ${builder.deviceId}: ${e}`));
+			const emulatorManager = new EmulatorManager(config);
+
+			await wakeDevices(logger, builder, emulatorManager)
+				.catch(e => logger.warn(`Could not wake ${builder.deviceId}: ${e}`));
+			await enableAutoRotation(logger, builder, emulatorManager)
+				.catch(e => logger.warn(`Could not enable auto-rotation ${builder.deviceId}: ${e}`));
 		}
 		done();
 	});
 };
 
-async function adb(argumentArray) {
-	return new Promise(resolve => appc.subprocess.run(ADB_PATH, argumentArray, { shell: false, windowsHide: true }, (error, stdout, _stderr) => {
-		resolve(stdout);
-	}));
+async function adb(args) {
+	return new Promise((resolve, reject) => {
+		let child;
+		let stdout = '';
+		let stderr = '';
+
+		if (process.platform === 'win32') {
+			child = spawn(
+				process.env.comspec || 'cmd.exe',
+				[ '/S', '/C', '"', ADB_PATH, ...args, '"' ],
+				{
+					windowsVerbatimArguments: true
+				}
+			);
+		} else {
+			child = spawn(
+				ADB_PATH,
+				args,
+				{
+					shell: false,
+					windowsHide: true
+				}
+			);
+		}
+		if (child) {
+			child.stdout.on('data', data => {
+				stdout += data.toString();
+			});
+			child.stderr.on('data', data => {
+				stderr += data.toString();
+			});
+			child.on('close', code => {
+				if (code !== 0) {
+					return reject(`${stdout}\n${stderr}`);
+				}
+				resolve(stdout);
+			});
+			return;
+		}
+		reject();
+	});
 }
 
-async function wakeDevices(logger, builder) {
+async function wakeDevices(logger, builder, emulatorManager) {
 
 	async function wake(device) {
 		logger.info(`Waking up ${device}`);
 
-		const deviceId = device !== 'emulator' ? [ '-s', device ] : [];
+		const deviceId = device !== 'emulator' ? [ '-s', await getDeviceId(device, emulatorManager) ] : [];
```

Comment:
Note that on the CI box it's spitting out `"Waking up android-30-playstore-x86"` - so presumably at this point `device === 'android-30-playstore-x86'`, which means it's actually going to trigger the code to `await getDeviceId(device, emulatorManager)` - which doesn't seem right to me. 

Looking at https://github.com/appcelerator/titanium_mobile/blob/master/android/cli/hooks/run.js#L85-L94

I think if `builder.target === 'emulator'` then `builder.deviceId` can be trusted?
So should this condition be testing if `builder.target === 'device'`? I'm not sure what you were trying to fix here by trying to get the deviceId in another way.

Reply:
You're right it should have been comparing `builder.target`, fixed that now.

The issue I was trying to fix is that it would use the incorrect device identifier for the emulator, which is an issue when multiple emulators are open.
```
[INFO]  Waking up Android_R_11.0_30_
[WARN]  Could not wake Android_R_11.0_30_:
adb.exe: more than one device/emulator
```
In this case, it would attempt to use the AVD identifier `Android_R_11.0_30_` instead of the device identifier `emulator-5554` to execute adb commands.

I guess this doesn't really matter on our test nodes, unless we want to test on multiple OS versions in the future. So I'll remove the handling of that for now.